modules/javafx.graphics/src/main/java/javafx/scene/image/PixelBuffer.java

```diff
@@ -0,0 +1,220 @@
+     */
+    public PixelBuffer(int width, int height, T buffer, PixelFormat<T> pixelFormat) {
+        Objects.requireNonNull(buffer, "buffer must not be null.");
+        Objects.requireNonNull(pixelFormat, "pixelFormat must not be null.");
+        if (width <= 0 || height <= 0) {
+            throw new IllegalArgumentException("PixelBuffer dimensions must be positive (w,h > 0)");
+        }
+        switch (pixelFormat.getType()) {
+            case BYTE_BGRA_PRE:
+                if (buffer.capacity() / width / 4 < height) {
+                    throw new IllegalArgumentException("Insufficient memory allocated for ByteBuffer.");
+                }
+                if (!(buffer instanceof ByteBuffer)) {
+                    throw new IllegalArgumentException("PixelFormat<ByteBuffer> requires a ByteBuffer.");
+                }
+                break;
+            case INT_ARGB_PRE:
+                if (buffer.capacity() / width < height) {
+                    throw new IllegalArgumentException("Insufficient memory allocated for IntBuffer.");
+                }
+                if (!(buffer instanceof IntBuffer)) {
+                    throw new IllegalArgumentException("PixelFormat<IntBuffer> requires an IntBuffer.");
+                }
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported PixelFormat: " + pixelFormat.getType());
+        }
+        this.buffer = buffer;
+        this.width = width;
+        this.height = height;
+        this.pixelFormat = pixelFormat;
+        this.imageRefs = new LinkedList<>();
+    }
+
+    /**
+     * Returns the {@link java.nio.Buffer} of this {@code PixelBuffer}.
+     *
+     * @return the {@link java.nio.Buffer} of this {@code PixelBuffer}
+     */
+    public T getBuffer() {
+        return buffer;
+    }
+
+    /**
+     * Returns the width of this {@code PixelBuffer}.
+     *
+     * @return the width of this {@code PixelBuffer}
+     */
+    public int getWidth() {
+        return width;
+    }
+
+    /**
+     * Returns the height of this {@code PixelBuffer}.
+     *
+     * @return the height of this {@code PixelBuffer}
+     */
+    public int getHeight() {
+        return height;
+    }
+
+    /**
+     * Returns the {@code PixelFormat} of this {@code PixelBuffer}.
+     *
+     * @return the {@code PixelFormat} of this {@code PixelBuffer}
+     */
+    public PixelFormat<T> getPixelFormat() {
+        return pixelFormat;
+    }
+
+    /**
+     * Invokes the specified {@link Callback} method and updates the dirty region of
+     * all {@code WritableImage}s that were created using this {@code PixelBuffer}.
+     * The {@code Callback} method is expected to update the {@code Buffer} and
+     * return a {@link Rectangle2D} that encloses the dirty region, or
+     * return {@code null} to indicate that entire buffer is dirty.
+     * This method must be called on the JavaFX Application thread.
+     * <p>
```

Comment:
I don't think there's a need for any `@link` in this method's doc since the classes are linked from the method's arguments.